### PR TITLE
[Reply] Animated contextual fab

### PR DESCRIPTION
**Pull request by Renzo-Olivares**
Animate fab based on context (either inbox page or mail view)

reply-fab-switcher

### CODE REVIEW

File: lib/studies/reply/adaptive_nav.dart [hunk 1]
```diff
@@ -824,7 +834,22 @@ class _ReplyFab extends StatelessWidget {
     } else {
       return FloatingActionButton(
         heroTag: 'Bottom App Bar FAB',
```
Comment (rami-a) [inline lib/studies/reply/adaptive_nav.dart:848]: The FAB should have a tooltip set for a11y. And the value should change depending on if on mail view or not

**Review by rami-a — APPROVED** [review]
LGTM

File: lib/studies/reply/adaptive_nav.dart (hunk 1 above)
Comment (Renzo-Olivares) [reply lib/studies/reply/adaptive_nav.dart:848]: Will add.

File: lib/studies/reply/mail_card_preview.dart [hunk 2]
```diff
@@ -1,10 +1,12 @@
 import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
```
Comment (guidezpl) [inline lib/studies/reply/mail_card_preview.dart]: nit: import sorting

File: lib/studies/reply/mail_view_page.dart [hunk 3]
```diff
@@ -1,7 +1,9 @@
 import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
```
Comment (guidezpl) [inline lib/studies/reply/mail_view_page.dart]: nit: import sorting

File: lib/studies/reply/adaptive_nav.dart [hunk 4]
```diff
@@ -824,7 +835,23 @@ class _ReplyFab extends StatelessWidget {
...
-        child: const Icon(Icons.create),
+        tooltip: onMailView ? 'Reply' : 'Compose',
+        child: AnimatedSwitcher(
+          duration: const Duration(milliseconds: 350),
+          transitionBuilder: (child, animation) => ScaleTransition(
+            child: child,
+            scale: animation,
+          ),
+          child: onMailView
+              ? Icon(
```
Comment (guidezpl) [inline lib/studies/reply/adaptive_nav.dart:857]: nit: would be more readable without the trailing commas after key

File: lib/studies/reply/adaptive_nav.dart [hunk 5]
```diff
@@ -1,12 +1,14 @@
 import 'dart:math' as math;
 import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
```
Comment (guidezpl) [inline lib/studies/reply/adaptive_nav.dart]: nit: import sorting

**Review by guidezpl — APPROVED** [review]
```
  _    ___ _____ __  __ 
 | |  / __|_   _|  \/  |
 | |_| (_ | | | | |\/| |
 |____\___| |_| |_|  |_|
                        
```

**Review by clocksmith — APPROVED** [review]
I really like the fab icon animation

**Pull request merged**